src/NLog/Targets/FastColoredConsoleTarget.cs

```diff
@@ -0,0 +1,174 @@
+        public FastColoredConsoleTarget()
+        {
+            _lastForegroundColor = Console.ForegroundColor;
+            _lastBackgroundColor = Console.BackgroundColor;
+            Error = false;
+            UseForegroundColor = true;
+            UseBackgroundColor = false;
+            UseResetColor = false;
+            TraceBackgroundColor = DebugBackgroundColor = InfoBackgroundColor =
+                WarningBackgroundColor = ErrorBackgroundColor = FatalBackgroundColor = ConsoleColor.Black;
+            TraceForegroundColor = ConsoleColor.DarkGray;
+            DebugForegroundColor = ConsoleColor.DarkGreen;
+            InfoForegroundColor = ConsoleColor.White;
+            WarningForegroundColor = ConsoleColor.Magenta;
+            ErrorForegroundColor = ConsoleColor.Yellow;
+            FatalForegroundColor = ConsoleColor.Red;
+
+        }
+        /// <summary>
+        /// Initializes the target.
+        /// </summary>
+        protected override void InitializeTarget()
+        {
+            base.InitializeTarget();
+            if (Header != null)
+            {
+                Output(Header.Render(LogEventInfo.CreateNullEvent()), LogLevel.Info);
+            }
+        }
+
+        /// <summary>
+        /// Closes the target and releases any unmanaged resources.
+        /// </summary>
+        protected override void CloseTarget()
+        {
+            if (Footer != null)
+            {
+                Output(Footer.Render(LogEventInfo.CreateNullEvent()), LogLevel.Info);
+            }
+            base.CloseTarget();
+        }
+
+        /// <summary>
+        /// Writes the specified logging event to the Console.Out or
+        /// Console.Error depending on the value of the Error flag.
+        /// </summary>
+        /// <param name="logEvent">The logging event.</param>
+        /// <remarks>
+        /// Note that the Error option is not supported on .NET Compact Framework.
+        /// </remarks>
+        protected override void Write(LogEventInfo logEvent)
+        {
+            Output(Layout.Render(logEvent), logEvent.Level);
+        }
+
+        private void Output(string s, LogLevel level)
+        {
+            if (UseForegroundColor || UseBackgroundColor)
+                if (level == LogLevel.Trace)
+                    SetColor(TraceForegroundColor, TraceBackgroundColor);
+                else if (level == LogLevel.Debug)
+                    SetColor(DebugForegroundColor, DebugBackgroundColor);
+                else if (level == LogLevel.Info)
+                    SetColor(InfoForegroundColor, InfoBackgroundColor);
+                else if (level == LogLevel.Warn)
+                    SetColor(WarningForegroundColor, WarningBackgroundColor);
+                else if (level == LogLevel.Error)
+                    SetColor(ErrorForegroundColor, ErrorBackgroundColor);
+                else if (level == LogLevel.Fatal)
+                    SetColor(FatalForegroundColor, FatalBackgroundColor);
```

Comment:
Please refactor this to a switch which sets two vars. (and the setColor call below the switch)